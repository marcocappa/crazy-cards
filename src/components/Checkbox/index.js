import React, { useState } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const Checkbox = ({
  className,
  name,
  value,
  isDisabled,
  label,
  isChecked,
  onChange,
}) => {
  const [checked, setChecked] = useState(isChecked);

  return (
    <>
      <input
        onChange={() => {
          onChange(value);
        }}
        type="checkbox"
        defaultChecked={checked}
        name={`${name}[]`}
        className={`crazy-card-checkbox ${classnames(className)}`}
        id={value}
        disabled={isDisabled}
      />
      <label
        htmlFor={value}
        onClick={() => (checked ? setChecked(false) : setChecked(true))}
      >
        {label}
      </label>
    </>
  );
};

Checkbox.propTypes = {
  className: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  isDisabled: PropTypes.bool,
  isChecked: PropTypes.bool,
};

Checkbox.defaultProps = {
  className: undefined,
  isDisabled: false,
  isChecked: false,
};

export default Checkbox;
