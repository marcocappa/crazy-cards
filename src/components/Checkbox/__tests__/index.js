import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../index';

describe('<Checkbox />', () => {
  const onChange = jest.fn();
  const props = {
    className: 'additional-class',
    name: 'list',
    value: 'student',
    isDisabled: false,
    label: 'Student Card',
    isChecked: false,
    onChange,
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Checkbox {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke onChange', () => {
    wrapper.find('input').simulate('change');
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(props.value);
  });
});
