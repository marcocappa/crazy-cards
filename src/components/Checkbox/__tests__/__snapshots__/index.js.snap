// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should match snapshot 1`] = `
<Fragment>
  <input
    className="crazy-card-checkbox additional-class"
    defaultChecked={false}
    disabled={false}
    id="student"
    name="list[]"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="student"
    onClick={[Function]}
  >
    Student Card
  </label>
</Fragment>
`;
